### PR TITLE
revamp workspace docs/tutorial to reflect new working dir changes

### PR DESCRIPTION
**Pull request by gibsondan**
Summary:
Also rename 'repository location' to 'code location' in code for now.

## Checklist

- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: docs/content/tutorial/advanced-tutorial/repositories.mdx [hunk 1]
```diff
@@ -60,17 +42,17 @@ Now you can see the list of all jobs in the repo on the left:
 <Image
 alt="repos.png"
 src="/images/tutorial/repos.png"
-width={4018}
-height={3062}
+width={3808}
+height={2414}
```
Comment (gibsondan) [inline docs/content/tutorial/advanced-tutorial/repositories.mdx:46]: this screenshot is not showing the right thing as per https://dagster.slack.com/archives/G01JPESLH1C/p1641855376003000 - happy to regenerate it but not sure how

File: docs/content/concepts/repositories-workspaces/workspaces.mdx [hunk 2]
```diff
@@ -32,7 +32,7 @@ The repositories in a workspace are loaded in a different process and communicat
 
 The structure of a workspace is encoded in a yaml document. By convention is it named `workspace.yaml`. The workspace yaml encodes where to find repositories.
 
-Each entry in the workspace is referred to as a _repository location_. A repository location can include more than one repository. Each repository location is loaded in its own process that Dagster tools use an RPC protocol to communicate with. This process separation allows multiple repository locations in different environments to be loaded independently, and ensures that errors in user code can't impact Dagster system code.
+Each entry in the workspace refers to a single code location, but that code location can include more than one repository. Each code location is loaded in its own process that Dagster tools use an RPC protocol to communicate with. This process separation allows multiple code locations in different environments to be loaded independently, and ensures that errors in user code can't impact Dagster system code.
```
Comment (sryza) [inline docs/content/concepts/repositories-workspaces/workspaces.mdx:35]: What's the reasoning behind changing repository location to code location? An advantage of repository location is that they can look it up in our docs if they want to learn more.

**Comment by gibsondan** [comment]
Current dependencies on/for this PR:
* master
  * **PR #6169**  (this PR)
    * **PR #6173** 

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh).

File: docs/content/concepts/repositories-workspaces/workspaces.mdx (hunk 2 above)
Comment (gibsondan) [reply docs/content/concepts/repositories-workspaces/workspaces.mdx:35]: we can split out that part into a separate discussion - agree that if we make the change we should probably do it everywhere at once.

Comment (gibsondan) [reply docs/content/concepts/repositories-workspaces/workspaces.mdx:35]: My thinking was that 'repository location' feels more like (yet another) term that you need to define, but a 'code location' is more self explanatory - it's just the location of your code.

Comment (sryza) [reply docs/content/concepts/repositories-workspaces/workspaces.mdx]: Cool yeah, I agree that it's more self-explanatory, but also worried about introducing yet another term.

File: docs/content/tutorial/advanced-tutorial/repositories.mdx [hunk 3]
```diff
@@ -60,17 +42,17 @@ Now you can see the list of all jobs in the repo on the left:
...
 />
 
 ## Workspaces
 
-Loading repositories via the `-f` or `-m` options is actually just a convenience function. The underlying abstraction is the "workspace", which determines all of the available repositories available to Dagit.
+Loading repositories via the `-f` option is actually just a convenience function. The underlying abstraction is the "workspace", which determines the set of repositories available to Dagit.
```
Comment (sryza) [inline docs/content/tutorial/advanced-tutorial/repositories.mdx:51]: Might be outside of the scope of the changes here, but part of the point of a workspace file is that it allows you to load multiple repository locations at once, right? Maybe worth mentioning.

**Review by sryza — APPROVED** [review]

**Pull request merged**